core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -890,7 +902,9 @@ private void applyConfiguration(CreateContainerCmd createCommand) {
             createCommand.withPrivileged(privilegedMode);
         }
 
-        createContainerCmdModifiers.forEach(hook -> hook.accept(createCommand));
+        this.createContainerCmdModifiers.forEach(createContainerCmdModifier -> {
+                createContainerCmdModifier.modify(createCommand);
+            });
```

Comment:
`modify`'s return value is ignored. Let's use the good old `for each` and re-assign `createCommand`